main.py

```diff
@@ -86,38 +86,41 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
              if not ok:
                 return
 
-        new_stream = {"url": stream_url, "quality": stream_quality}
-
+        streams = None
         try:
-            self.setup_videoframe(new_stream, self.coordinates)
-            self.new_coordinates()
-        except KeyError:
+            session = Streamlink()
+            streams = session.streams(stream_url)
+        except streamlink_exceptions.NoPluginError:
+            error_window = QtWidgets.QMessageBox().warning(self,
+                "Error",
+                "Could not open stream: The provided URL is not supported"
+            )
+
+            self.add_new_stream()
+            return
+        
+        if stream_quality not in streams:
```

Comment:
This, and the following lines below feels very hard to read imo and should probably be broken up into some other function or something right? It feels kind of unnatural and jumpy to read with the return statement above, maybe we can refactor the layout of the function some how?